src/lib/userStorage/UserProfileStorage.js

```diff
@@ -0,0 +1,556 @@
+   * @returns
+   */
+  getProfile(): { [key: string]: string } {
+    return Object.keys(this.profile).reduce((acc, currKey) => ({ ...acc, [currKey]: this.profile[currKey].value }), {})
+  }
+
+  /**
+   * updates profile fields in realm
+   * @param {*} fields
+   * @returns
+   */
+  async setProfileFields(fields: { [key: string]: ProfileField }): Promise<any> {
+    const encryptedFields = await this._encryptProfileFields(fields)
+    await this.profiledb.setProfileFields(encryptedFields)
+    this._setLocalProfile({ ...this.profile, ...fields })
+  }
+
+  /**
+   * create new profile with given fields in realm
+   * @param fields
+   * @returns {Promise<void>}
+   * @private
+   */
+  async setNewProfileFields(fields: { [key: string]: ProfileField }): Promise<any> {
+    const encryptedFields = await this._encryptProfileFields(fields)
+    await this.profiledb.setProfile(encryptedFields)
+    this._setLocalProfile({ ...this.profile, ...fields })
+  }
+
+  /**
+   * helper to set display field based on privacy setting
+   * @param field
+   * @param value
+   * @param privacy
+   * @returns {*}
+   * @private
+   */
+  _setDisplayFieldBasedOnPrivacy(field: string, value: string, privacy: string) {
+    let display
+    switch (privacy) {
+      case 'private':
+        display = '******'
+        break
+      case 'masked':
+        display = UserProfileStorage.maskField(field, value)
+
+        //undo invalid masked field
+        if (display === value) {
+          privacy = 'public'
+        }
+        break
+      case 'public':
+        display = value
+        break
+      default:
+        throw new Error('Invalid privacy setting', { privacy })
+    }
+    return display
+  }
+
+  /**
+   * Set profile field with privacy
+   * @param {*} field
+   * @param {*} value
+   * @param {*} privacy
+   * @param {*} onlyPrivacy
+   * @returns
+   */
+  setProfileField(
+    field: string,
+    value: string,
+    privacy: FieldPrivacy = 'public',
+    onlyPrivacy: boolean = false,
+  ): Promise<ACK> {
+    const display = this._setDisplayFieldBasedOnPrivacy(field, value, privacy)
+
+    logger.debug('setProfileField', { field, value, privacy, onlyPrivacy, display })
+
+    return this.setProfileFields({ [field]: { display, value, privacy } })
+  }
+
+  /**
+   * Avatar setter
+   * @returns {Promise<CID[]>}
+   */
+  async setAvatar(avatar) {
+    const cids = await this._resizeAndStoreAvatars(avatar)
+
+    await Promise.all(
+      // eslint-disable-next-line require-await
+      toPairs(cids).map(async ([field, value]) => this.setProfileField(field, value, 'public')),
+    )
+  }
+
+  /**
+   * Remove field from db
+   * @param field
+   * @returns {Promise<*>}
+   */
+  // eslint-disable-next-line require-await
+  async removeField(field: string) {
+    const profileToUpdate = this.profile
+    profileToUpdate[field] = { value: null, display: null, privacy: this.getFieldPrivacy(field) }
```

Comment:
also please do not mutate this.profile. 
make new object copying the current profile via spread operator

```javascript
const privacy = this.getFieldPrivacy(field)

const profileToUpdate = {
  ...this.profile,
  [field]: { value: null, display: null, privacy }
}
```